src/main/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -334,6 +334,19 @@ private boolean hasReferenceName() {
     }
 
     /**
+     * @return {@code true} if records belong to the same pairwise alignment
+     */
+    public boolean isPair(final SAMRecord record) {
```

Comment:
I'm confused about this function, a mate is another record with the same query name...why are you checking everything else, and not the QNAME? I'm sure there's a reason, but it should be explained in the javadoc as it is not obvious at all!

Reply:
Thank you for the remark, I think it does matter in `SamRecord` and I fixed it, but if we implement it in `HitsForInsert` it doesn't matter because it deals records with the same `QNAME` only. [Here](https://github.com/broadinstitute/picard/blob/master/src/main/java/picard/sam/MultiHitAlignedReadIterator.java#L115) is the logic that produces HitsForInsert.

Reply:
I would suggest to move this to `SamPairUtil`, because there are already a lot of methods in `SAMRecord` that may be better to be used within an utility class. Eventually, reads would be accessed by an interface (e.g., #985) and thus this methods may live in an utility class that is implementation-independent.